### ISSUE TITLE
How to correctly load a private key

### ISSUE THREAD

**Issue by iulianpascalau** [dchest/tweetnacl-js#177]
Hello
We have an issue trying to match generated public keys from tweetnacl-js and kyber-go lib. Starting from the same private key in hex format (example: `a40e45c66bc9506b65efc676a9d84a4df395eeac30db620679fab1748d6e7201`) we yield different results (using tweetnacl-js: `15c66e71a37000f0f71cc917568a5e34d7d418aa0a07ed4af295b1f82a3d484d` and on kyber-go: `8a2ee461bd72652fc33d8705b33cf240dc4a1531c5bf80bd4f4d92b6d83636ae`) 
What intrigues me is that the signing/verifying process works between those 2 implementations (a string signed with tweetnacl-js passes the verification on kyber-go).
In tweetnacl-js we have tried both `nacl.sign.keyPair.fromSeed` and `nacl.sign.keyPair.fromSecretKey` functions. The later, was fed with the concatenation between the private key and the kyber-generated public key but the resulting signature was invalid.
I guess I am doing something wrong.
```JS
const hexPrivate = Buffer.from('a40e45c66bc9506b65efc676a9d84a4df395eeac30db620679fab1748d6e7201', 'hex');
const kp = tweetnacl.sign.keyPair.fromSeed(privateKey);
const pk = kp.publicKey
const sk = kp.secretKey
const skHex = Buffer.from(sk).toString('hex')
const pkHex = Buffer.from(pk).toString('hex')

console.log(skHex, ' ', pkHex)
```

**Comment by dchest** [comment]
Perhaps, it's better to ask kyber-go authors, since this package uses the original NaCl/TweetNaCl Ed25519 key formats. I'm not sure what kyber-go does.

**Comment by iulianpascalau** [comment]
Ok, we have sorted things out. tweetnacl-js and https://godoc.org/golang.org/x/crypto/ed25519 comply with rfc8032 and actually store the seed as private key. When generating the public key, the libs do a sha512 on the seed and apply some bit masking. Again, whenever a sign func is called, libs also do a hash on the seed. Our kyber-based implementation stores the private key's raw bytes (not the seed) and we skipped the hashing calls so that is why we are getting another pair of secret-public keys. 
Thanks for pointing us in the right direction!
